pkg/apis/kudo/v1alpha1/instance_types.go

```diff
@@ -16,65 +16,372 @@ limitations under the License.
+func (i *Instance) EnsurePlanStatusInitialized(ov *OperatorVersion) {
+	if i.Status.PlanStatus == nil {
+		i.Status.PlanStatus = make(map[string]PlanStatus)
+	}
+
+	for planName, plan := range ov.Spec.Plans {
+		planStatus := &PlanStatus{
+			Name:   planName,
+			Status: ExecutionNeverRun,
+			Phases: make([]PhaseStatus, 0),
+		}
+
+		existingPlanStatus, planExists := i.Status.PlanStatus[planName]
+		if planExists {
+			planStatus.Status = existingPlanStatus.Status
+		}
+		for _, phase := range plan.Phases {
+			phaseStatus := &PhaseStatus{
+				Name:   phase.Name,
+				Status: ExecutionNeverRun,
+				Steps:  make([]StepStatus, 0),
+			}
+			existingPhaseStatus, phaseExists := PhaseStatus{}, false
+			if planExists {
+				for _, oldPhase := range existingPlanStatus.Phases {
+					if phase.Name == oldPhase.Name {
+						existingPhaseStatus = oldPhase
+						phaseExists = true
+						phaseStatus.Status = existingPhaseStatus.Status
+					}
+				}
+			}
+			for _, step := range phase.Steps {
+				stepStatus := StepStatus{
+					Name:   step.Name,
+					Status: ExecutionNeverRun,
+				}
+				if phaseExists {
+					for _, oldStep := range existingPhaseStatus.Steps {
+						if step.Name == oldStep.Name {
+							stepStatus.Status = oldStep.Status
+						}
+					}
+				}
+				phaseStatus.Steps = append(phaseStatus.Steps, stepStatus)
+			}
+			planStatus.Phases = append(planStatus.Phases, *phaseStatus)
+		}
+		i.Status.PlanStatus[planName] = *planStatus
+	}
+}
+
+// StartPlanExecution mark plan as to be executed
+func (i *Instance) StartPlanExecution(planName string, ov *OperatorVersion) error {
+	if i.NoPlanEverExecuted() || isUpgradePlan(planName) {
+		i.EnsurePlanStatusInitialized(ov)
+	}
+
+	// update status of the instance to reflect the newly starting plan
+	notFound := true
+	for n1, v := range i.Status.PlanStatus {
+		if v.Name == planName {
+			// update plan status
+			notFound = false
+			planStatus := i.Status.PlanStatus[n1]
+			planStatus.Status = ExecutionPending
+			for i2, p := range v.Phases {
+				planStatus.Phases[i2].Status = ExecutionPending
+				for i3 := range p.Steps {
+					i.Status.PlanStatus[n1].Phases[i2].Steps[i3].Status = ExecutionPending
+				}
+			}
+
+			i.Status.PlanStatus[n1] = planStatus // we cannot modify item in map, we need to reassign here
+
+			// update activePlan and instance status
+			i.Status.AggregatedStatus.Status = ExecutionPending
+			i.Status.AggregatedStatus.ActivePlanName = planName
+
+			break
+		}
+	}
+	if notFound {
+		return &InstanceError{fmt.Errorf("asked to execute a plan %s but no such plan found in instance %s/%s", planName, i.Namespace, i.Name), kudo.String("PlanNotFound")}
+	}
 
-// PhaseState captures the state of the rollout.
-type PhaseState string
+	// TODO in the future when we again support manual plan execution, snapshot should be saved only non non-manually executed plans
```

Comment:
```suggestion
	// TODO in the future when we again support manual plan execution, snapshot should be saved only manually executed plans
```
could we state this in a positive way instead of a double negative?